fossunited/chapters/doctype/foss_event_rsvp_submission/foss_event_rsvp_submission.py

```diff
@@ -33,9 +33,46 @@ class FOSSEventRSVPSubmission(Document):
     # end: auto-generated types
     pass
 
+    def before_insert(self):
+        self.validate_form_published()
+
     def validate(self):
         self.validate_linked_rsvp_exists()
 
+    def after_insert(self):
+        self.close_rsvp_on_max_count()
+
+    def close_rsvp_on_max_count(self):
+        max_count = self.get_max_count()
+        submission_count = frappe.db.count(
+            "FOSS Event RSVP Submission",
+            {"linked_rsvp": self.linked_rsvp},
+        )
+
+        if submission_count >= max_count:
+            frappe.db.set_value(
+                "FOSS Event RSVP",
+                self.linked_rsvp,
+                "is_published",
+                False,
+            )
+
+    def get_max_count(self):
+        max_count = frappe.db.get_value(
+            "FOSS Event RSVP", self.linked_rsvp, "max_rsvp_count"
+        )
+        return max_count
+
+    def validate_form_published(self):
```

Comment:
This might not be needed. 

Since this is a bit redundant as there are other methods that reflect an unpublish with a UI change and permission levels in the submission doctype that blocks a direct insert by client API.